calyx-py/test/correctness/queues/binheap/pifo.py

```diff
@@ -125,14 +125,16 @@ def binheap_invoke_helper(value, rank):
         cb.if_with(
             err_eq_0,
             [
-                infer_flow_out,
                 cb.if_with(
                     cmd_eq_0,
-                    cb.if_with(
-                        turn_eq_flow_out,
-                        comp.incr(turn),
-                        cb.if_(flow_out.out, r_a_incr_2, r_b_incr_2),
-                    ),
+                    [
+                        infer_flow_out,
```

Comment:
I think I found a case where you were inferring the out-flow no matter the command, but you obviously only need to do that when the command was a pop. This little change fixes that.